File: wallet/src/main/java/chatox/wallet/service/UserService.java
package chatox.wallet.service;

import chatox.wallet.api.response.UserResponse;
import chatox.wallet.model.User;

import java.util.Optional;

public interface UserService {
    Optional<User> findUserById(String id);
    User requireUserById(String id);
    void saveUser(UserResponse userResponse);
}

File: chat/src/main/kotlin/chatox/chat/service/UserService.kt
package chatox.chat.service

import chatox.chat.api.response.UserResponse
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface UserService {
    fun findUserById(id: String): Mono<UserResponse>
    fun findUserByIdAndPutInLocalCache(id: String?, localCache: MutableMap<String, UserResponse>? = null): Mono<UserResponse>
    fun findAllByIdAndPutInLocalCache(ids: List<String>, localCache: MutableMap<String, UserResponse>? = null): Flux<UserResponse>
}
